src/components/cards/Canvas.tsx

```diff
@@ -13,18 +15,18 @@ const NodeTypes = {
 };
 
 const CardCanvas = ({ data, blockHeight, columns }: any) => {
-  const initialEdges = useMemo(() => parseEdges(data), [data]);
-  const initialNodes = useMemo(
-    () =>
-      parseNodes(data, {
+  const selectedEntries = useSelectedEntries(data);
```

Comment:
I would move the usage of `useFilter` into the block instead of the canvas

Reply:
yes